### PR TITLE
Species languages

### PR DESCRIPTION
**Pull request by d3athrow**

### CODE REVIEW

**Comment by SkowronX** [comment]
please

**Comment by mrpain666** [comment]
ITS HABBENING

**Comment by CptWad** [comment]
thank you :+1:

File: code/game/gamemodes/heist/heist.dm [hunk 1]
```diff
@@ -88,9 +88,10 @@
 		vox.dna.mutantrace = "vox"
 		vox.set_species("Vox")
 		vox.generate_name()
-		vox.languages = HUMAN // Removing language from chargen.
+		//vox.languages = HUMAN // Removing language from chargen.
 		vox.flavor_text = ""
-		//vox.add_language("Vox-pidgin")
+		vox.default_language = all_languages["Vox-pidgin"]
```
Comment (ComicIronic) [inline code/game/gamemodes/heist/heist.dm]: Are you going to use defines for the language names?

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 2]
```diff
@@ -23,6 +23,63 @@
...
+	for(var/language in languages)
+		mind.changeling.absorbed_languages |= language
+	return 1
+
+//Used to dump the languages from the changeling datum into the actual mob.
+/mob/proc/changeling_update_languages(var/updated_languages)
+
+	languages = list()
+	for(var/language in updated_languages)
+		languages += language
```
Comment (ComicIronic) [inline code/game/gamemodes/changeling/changeling_powers.dm]: If I understand it right, shouldn't this add_language(language.name) in the same standard as you set before?

File: code/game/gamemodes/changeling/changeling_powers.dm [hunk 3]
```diff
@@ -904,11 +983,13 @@ var/list/datum/dna/hivemind_bank = list()
 	if(!changeling)
 		return 0
 
-	var/mob/living/carbon/T = changeling_sting(40, /mob/proc/changeling_extract_dna_sting)
+	var/mob/living/carbon/human/T = changeling_sting(40, /mob/proc/changeling_extract_dna_sting)
 	if(!T)	return 0
 
 	T.dna.real_name = T.real_name
 	changeling.absorbed_dna |= T.dna
+	if(T.species && !(T.species.name in changeling.absorbed_species))
```
Comment (ComicIronic) [inline code/game/gamemodes/changeling/changeling_powers.dm]: You could just do

    if(T.species)
         changeling.absorbed_species |= T.species.name

right?

File: code/game/objects/items/devices/taperecorder.dm [hunk 4]
```diff
@@ -22,7 +22,7 @@
 	throw_speed = 4
 	throw_range = 20
 
-/obj/item/device/taperecorder/Hear(message, atom/movable/speaker, message_langs, raw_message, radio_freq)
+/obj/item/device/taperecorder/Hear(message, atom/movable/speaker, var/datum/language/speaking, raw_message, radio_freq)
```
Comment (ComicIronic) [inline code/game/objects/items/devices/taperecorder.dm]: Now we just need the recorder to store the message language.

File: code/modules/mob/dead/observer/observer.dm [hunk 5]
```diff
@@ -653,7 +655,7 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 
 	if(host)
 		if(config.uneducated_mice)
-			languages = NONE
+			host.universal_understand = 0
```
Comment (ComicIronic) [inline code/modules/mob/dead/observer/observer.dm]: Do mice have universal_speak set to 0 as well?

File: code/modules/mob/living/carbon/alien/larva/larva.dm [hunk 6]
```diff
@@ -22,6 +22,7 @@
 		name = "alien larva ([rand(1, 1000)])"
 	real_name = name
 	regenerate_icons()
+	add_language("Xenomorph")
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/alien/larva/larva.dm]: Do they not start with it as a default_language?

File: code/modules/mob/living/silicon/ai/ai.dm [hunk 7]
```diff
@@ -68,7 +68,14 @@ var/list/ai_list = list()
 			if (A.real_name == pickedName && possibleNames.len > 1) //fixing the theoretically possible infinite loop
 				possibleNames -= pickedName
 				pickedName = null
-
+	add_language("Robot Talk", 1)
```
Comment (ComicIronic) [inline code/modules/mob/living/silicon/ai/ai.dm]: Why not just use universal_speak and universal_understand?

**Comment by CptWad** [comment]
Cant hear myself say anything as vox. Including vox and reg language

File: code/modules/mob/dead/observer/observer.dm (hunk 5 above)
Comment (d3athrow) [inline code/modules/mob/dead/observer/observer.dm]: Yes they should.

File: code/modules/mob/living/carbon/alien/larva/larva.dm (hunk 6 above)
Comment (d3athrow) [inline code/modules/mob/living/carbon/alien/larva/larva.dm]: default_language is a string.

File: code/modules/mob/living/silicon/ai/ai.dm (hunk 7 above)
Comment (d3athrow) [inline code/modules/mob/living/silicon/ai/ai.dm]: Actually don't know why i added this as its not a language. Its meant for binary actually.

**Comment by CptWad** [comment]
Hey theres no special colors for vox speach and maybe others like before. That intended?

**Comment by d3athrow** [comment]
Vox talk has color though, i used and saw it in the test serb

**Comment by IratePirate** [comment]
>[content]

Fixing something some asshat broke is considered content?

**Comment by d3athrow** [comment]
a port replaced it with an inferior feature, but it was working.

**Comment by 9600bauds** [comment]
I once told you I would give you 2 CS:GO keys when languages were fixed and I will honor this promise

File: code/game/gamemodes/changeling/changeling_powers.dm (hunk 3 above)
Comment (d3athrow) [inline code/game/gamemodes/changeling/changeling_powers.dm]: Checking it beforehand is less costly than doing |=

**Comment by mrpain666** [comment]
are pai's able to translate yet?

**Pull request merged**